ocs_ci/utility/tests/test_version.py

```diff
@@ -0,0 +1,49 @@
+from semantic_version import Version
+
+import pytest
+
+from ocs_ci.utility import version
+
+
+@pytest.mark.parametrize(
+    "ocs_version, only_major_minor, ignore_pre_release, expected",
+    [
+        ("4.4", False, False, Version.coerce("4.4")),
+        ("4.4.5", True, False, version.VERSION_4_4),
+        ("4.4.5", True, True, version.VERSION_4_4),
+        ("4.4.5-182.ci", True, False, version.VERSION_4_4),
+        ("4.4.5-182.ci", False, False, Version.coerce("4.4.5-182.ci")),
+    ],
+)
+def test_get_semantic_version(
+    ocs_version, only_major_minor, ignore_pre_release, expected
+):
+    tested_version = version.get_semantic_version(
+        ocs_version, only_major_minor, ignore_pre_release
+    )
+    assert tested_version == expected
+
+
+@pytest.mark.parametrize(
+    "ocs_version, only_major_minor, ignore_pre_release, expected",
+    [
+        ("4.4", False, False, "4.4.0"),
+        ("4.4.5", True, False, "4.4"),
+        ("4.4.5", True, True, "4.4"),
+        ("4.4.5-182.ci", True, False, "4.4"),
+        ("4.4.5-182.ci", False, False, "4.4.5-182.ci"),
+        ("4.4.5-182.ci", False, True, "4.4.5"),
+    ],
+)
+def test_get_semantic_version_string_values(
+    ocs_version, only_major_minor, ignore_pre_release, expected
+):
+    tested_version = version.get_semantic_version(
+        ocs_version, only_major_minor, ignore_pre_release
+    )
+    assert f"{tested_version}" == expected
```

Comment:
Could this be written just as `tested_version == expected`, or is there some string formatting magic going on?

Reply:
This test is about the string formatting of the semantic version. As I pointed out in the meeting. If you will do
```python
 f"{Version.coerce("4.4")}" # this will be formatted as "4.4.0"
```
But when I am using only_major_minor which I am setting up patch and pre_release versions to None, then the string formatting will only be `"4.4"` when I am using only_major_minor=True.

Hence I am testing this behavior it's really working as expected when formatting semantic version as a string.

I cannot compare string value and semantic version object as those are completely different objects. Hence here have to compare string with string.

I can use the str(tested_version) as well, but I wanted to show the example how user will use the formatting of the semantic version which will most likely be used via formatting which at the end anyway call the __str__ on the tested_object.

So if you want, I can change to `str` instead but then I thought that it will not show the example I intended to show here.  @mbukatov let me know please if you don't mind to keep it as is, or if you think we should really use `str` instead.

Reply:
I would still argue that using `str()` is more readable in your case as well, but I don't think we need to change it.